tests/rbd/rbd_tier0.py

```diff
@@ -0,0 +1,317 @@
+import json
+import re
+from datetime import datetime
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.krbd_io_handler import krbd_io_handler
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_rbd(rbd, pool_type, **kw):
+    """Verify tier-0 RBD operations.
+
+    This module verifies tier-0 RBD operations
+
+    Args:
+        kw: test data
+
+    Examples:
+        kw:
+            config:
+                ec-pool-k-m: 2,1
+               ec-pool-only: False
+               ec_pool_config:
+                  pool: rbd_pool
+                  data_pool: rbd_ec_pool
+                  ec_profile: rbd_ec_profile
+                  image: rbd_image
+                  image_thick_provision: rbd_thick_image
+                  thick_size: 2G
+                  size: 10G
+                  snap: rbd_ec_pool_snap
+                  clone: rbd_ec_pool_clone
+               rep_pool_config:
+                  pool: rbd_rep_pool
+                  image: rbd_rep_image
+                  image_thick_provision: rbd_rep_thick_image
+                  thick_size: 2G
+                  size: 10G
+                  snap: rbd_rep_pool_snap
+                  clone: rbd_rep_pool_clone
+               operations:
+                  map: true
+                  io: true
+                  nounmap: false
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+    """
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        size = kw["config"][pool_type]["size"]
+        size_int = int(re.findall(r"\d+", size)[0])
+        snap = kw["config"][pool_type].get("snap", f"{image}_snap")
+        clone = kw["config"][pool_type].get("clone", f"{image}_clone")
+        image_spec = f"{pool}/{image}"
+        snap_name = f"{pool}/{image}@{snap}"
+
+        # Run fio and krbd map operations on image
+        kw.get("config")["image_spec"] = [image_spec]
+        kw.get("config")["size"] = size
+        kw["rbd_obj"] = rbd
+        krbd_io_handler(**kw)
+
+        # List images in a pool and verify
+        image_list = rbd.list_images(pool)
+        if image not in image_list:
+            log.error(f"Image {image} in not present in images listed for pool {pool}")
+            return 1
+
+        # Display image info and verify
+        image_info = rbd.image_info(pool, image)
+        log.info(f"Image info : {image_info}")
+        if image not in image_info["name"]:
+            log.error(f"Image {image} in not listed for pool {pool}")
```

Comment:
```suggestion
            log.error(f"Image {image} is not listed for pool {pool}")
```
Also, it would be great if we an tue error message based on the operation